lib/chef/application/client.rb

```diff
@@ -480,7 +480,7 @@ def interval_sleep(sec)
   end
 
   def client_sleep(sec)
-    IO.select([ SELF_PIPE[0] ], nil, nil, sec) or return
```

Comment:
This is actually the correct idiomatic usage of `or`, for flow control instead of joining multiple conditions.